grammar.js

```diff
@@ -608,6 +609,36 @@ module.exports = grammar({
 
     pointer_type: $ => prec(PREC.POSTFIX, seq($._type, '*')),
 
+    function_pointer_type: $ => seq(
+      'delegate',
+      '*',
+      optional($.calling_convention_specifier),
+      '<',
+      commaSep1($.funcptr_parameter),
+      '>'
+    ),
+
+    calling_convention_specifier: $ => choice(
```

Comment:
Roslyn calls this `FunctionPointerCallingConvention` so perhaps `function_pointer_calling_convention` ?